commons/all/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java

```diff
@@ -1,153 +1,8 @@
-
-
-      // Util
-      SYS_ALLOWED_CLASSES.add(Date.class.getName());
-
-      // Misc
-      SYS_ALLOWED_CLASSES.add(Enum.class.getName());
-      SYS_ALLOWED_CLASSES.add(Number.class.getName());
-
-      // Reference array regex, for arrray representations of allowed classes e.g '[Ljava.lang.Byte;'
-      SYS_ALLOWED_REGEXP.add("^\\[[\\[L].*\\;$");
-
-      String regexps = System.getProperty(REGEXPS_PROPERTY_NAME);
-      if (regexps != null) {
-         SYS_ALLOWED_REGEXP.addAll(asList(regexps.trim().split(",")));
-      }
-      String cls = System.getProperty(CLASSES_PROPERTY_NAME);
-      if (cls != null) {
-         SYS_ALLOWED_CLASSES.addAll(asList(cls.trim().split(",")));
-      }
-   }
-
-   private final Set<String> classes = new CopyOnWriteArraySet<>(SYS_ALLOWED_CLASSES);
-   private final List<String> regexps = new CopyOnWriteArrayList<>(SYS_ALLOWED_REGEXP);
-   private final List<Pattern> compiled = new CopyOnWriteArrayList<>();
-
-   public ClassWhiteList() {
-      this(Collections.emptySet(), Collections.emptyList());
-   }
-
-   public ClassWhiteList(List<String> regexps) {
-      this(Collections.emptySet(), regexps);
-   }
-
-   public ClassWhiteList(Collection<String> classes, List<String> regexps) {
-      Collection<String> classList = requireNonNull(classes, "Classes must not be null");
-      Collection<String> regexList = requireNonNull(regexps, "Regexps must not be null");
-      this.classes.addAll(classList);
-      this.regexps.addAll(regexList);
-      this.compiled.addAll(this.regexps.stream().map(Pattern::compile).collect(Collectors.toList()));
-   }
-
-   public boolean isSafeClass(String className) {
-      // Test for classes first (faster)
-      boolean isClassAllowed = classes.contains(className);
-      if (isClassAllowed) return true;
-      boolean regexMatch = compiled.stream().anyMatch(p -> p.matcher(className).find());
-      if (regexMatch) {
-         // Add the class name to the classes set to avoid future regex checks
-         classes.add(className);
-         return true;
-      }
-      if (log.isTraceEnabled())
-         log.tracef("Class '%s' not in whitelist", className);
-      return false;
-   }
-
-   public void addClasses(Class<?>... classes) {
-      stream(classes).forEach(c -> this.classes.add(c.getName()));
-   }
-
-   public void addClasses(String... classes) {
-      this.classes.addAll(Arrays.asList(classes));
-   }
-
-   public void addRegexps(String... regexps) {
-      this.regexps.addAll(asList(regexps));
-      this.compiled.addAll(stream(regexps).map(Pattern::compile).collect(Collectors.toList()));
-   }
-
-   public void read(ClassWhiteList whiteList) {
-      this.regexps.addAll(whiteList.regexps);
-      this.compiled.addAll(whiteList.compiled);
-      this.classes.addAll(whiteList.classes);
-   }
+ * @deprecated since 12.0. Will be removed in 14.0. Use {@link ClassAllowList}.
+ **/
+@Deprecated
+public final class ClassWhiteList extends ClassAllowList {
```

Comment:
We still need to keep the old constructors for backwards compatibility:

```java
   public ClassWhiteList() {
      super();
   }

   public ClassWhiteList(List<String> regexps) {
      super(regexps);
   }

   public ClassWhiteList(Collection<String> classes, List<String> regexps) {
      super(classes, regexps);
   }
```

Reply:
Done